installinator/src/fetch.rs

```diff
@@ -307,10 +299,16 @@ impl FetchArtifactBackend {
     }
 }
 
+/// Backend implementation for fetching artifacts.
+///
+/// Note: While [`crate::reporter::ReportProgressImpl`] is a persistent
+/// structure, a new `FetchArtifactImpl` is generated separately each time
+/// discovery occurs. We should align this with `ReportProgressImpl` in the
+/// future, though we'd need some way of looking up delay information in
+/// `mock_peers`.
 #[async_trait]
 pub(crate) trait FetchArtifactImpl: fmt::Debug + Send + Sync {
-    fn peers(&self) -> Box<dyn Iterator<Item = PeerAddress> + Send + '_>;
-    fn peer_count(&self) -> usize;
+    fn peers(&self) -> PeerAddresses;
```

Comment:
Could this be

```suggestion
    fn peers(&self) -> &PeerAddresses;
```

and let the caller clone it if desired?

Reply:
Ah actually -- the mock/fake implementation generates a `PeerAddresses` on the fly. We could store the addresses in the required form, I guess.